core/src/test/java/org/apache/calcite/test/SqlFunctionsTest.java

```diff
@@ -810,6 +824,642 @@ private void thereAndBack(byte[] bytes) {
     assertThat(SqlFunctions.multisetUnionDistinct(z, addc),
         is(Arrays.asList("a", "c", "d")));
   }
+
+  @Test public void testJsonValueExpression() {
+    assertThat(SqlFunctions.jsonValueExpression("{}"), is(Collections.emptyMap()));
+  }
+
+  @Test public void testJsonStructuredValueExpression() {
+    assertThat(SqlFunctions.jsonStructuredValueExpression("bar"),
+        is("bar"));
+    assertThat(SqlFunctions.jsonStructuredValueExpression(100),
+        is(100));
+  }
+
+  @Test public void testJsonApiCommonSyntax() {
+    assertThat(
+        SqlFunctions.
+            jsonApiCommonSyntax(ImmutableMap.of("foo", "bar"),
+                "lax $.foo").pathReturned,
+        is("bar"));
```

Comment:
Please factor that to `assertJsonApiCommonSyntax_pathReturned` method so the exception message contains input, expected and actual values.

Reply:
+1 thanks.

Reply:
@vlsi Thanks, I see your comments both in PR and JIRA, I will try do an optimization.